### ISSUE TITLE
DisallowVariableParsing: disallowSimpleSyntax false positive for `{$a->test($b)}`

### ISSUE THREAD

**Issue by LastDragon-ru** [slevomat/coding-standard#1559]
```xml
<?xml version="1.0"?>
<ruleset name="LaraAspStandard"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="./vendor/squizlabs/php_codesniffer/phpcs.xsd">
    <rule ref="SlevomatCodingStandard.Strings.DisallowVariableParsing">
        <properties>
            <property name="disallowSimpleSyntax" value="true"/>
            <property name="disallowDollarCurlySyntax" value="true"/>
            <property name="disallowCurlyDollarSyntax" value="false"/>
        </properties>
    </rule>
</ruleset>
```
```php
<?php declare(strict_types = 1);

namespace tmp;

class Bug {
    public function test(string $string = __FUNCTION__): string {
        return $string;
    }

    public static function bug(): void {
        $a = new Bug();
        $b = __FUNCTION__;

        echo "{$a->test()}";        // ok
        echo "{$a->test($b)}";      // error -> false positive
        echo "{$a->test('test')}";  // ok
    }
}
```
```txt
------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------
 15 | ERROR | Using variable syntax "$..." inside string is disallowed, found "$b".
------------------------------------------------------------------------------------
```
